metadata-ingestion/src/datahub/ingestion/source/nifi.py

```diff
@@ -196,6 +197,73 @@ def validator_site_url(cls, site_url: str) -> str:
         return site_url
 
 
+class BidirectionalComponentGraph:
+    def __init__(self):
+        self.outgoing: Dict[str, Set[str]] = defaultdict(set)
+        self.incoming: Dict[str, Set[str]] = defaultdict(set)
+        # this will not count duplicates/removal of non-existing connections correctly - it is only for quick check
+        self.connections_cnt = 0
```

Comment:
Can we rename instance variables incoming, outgoing and connections_cnt to start with an underscore to clarify that they are not to be modified/accessed outside of class ?

Reply:
Renamed the variables, as requested. Do we follow this pattern in other relevant classes as well? I haven't noticed many fields starting with a `_`.